auslib/test/test_db.py

```diff
@@ -2228,6 +2230,33 @@ def testGetRulesMatchingQueryVersionComparison(self):
         rules = self._stripNullColumns(rules)
         self.assertEquals(rules, [])
 
+    def testGetRulesMatchingQueryListOfVersionsComparison(self):
+        expected = [dict(rule_id=4, priority=90, backgroundRate=100,
+                         version='3.0.1,3.0.2,3.0b3', update_type='z', data_version=1)]
+        for version in ['3.0.1', '3.0.2', '3.0b3']:
+            rules = self.rules.getRulesMatchingQuery(
+                dict(name='', product='', version=version, channel='',
+                     buildTarget='', buildID='', locale='', osVersion='',
+                     distribution='', distVersion='', headerArchitecture='',
+                     force=False, queryVersion=3,
+                     ),
+                fallbackChannel=''
+            )
+            rules = self._stripNullColumns(rules)
+            self.assertEquals(rules, expected)
+
+        # version number isn't present in the list of version numbers.
```

Comment:
Very good job on these tests - you're covering most the cases. Small nit here: this test of the negative case should be a separate test. Generally, you should only be calling the code under test once per test method (the loop above is OK because you are testing the same expected result, just with different input values).

One other case to test too: ensuring that a partial version match doesn't match a longer version. Eg: if the version is set to "30.0.1,30.0.2,30.0.2", passing a version of "30.0" to getRulesMatchingQuery shouldn't come up with a match.